### PR TITLE
Make AllMetadataAction filterable

### PR DESCRIPTION
**Pull request by adrw**
- [x] Change existing config/webaction endpoints in Misk-Web and web action to not conflict with this
- [x] Add tests

Screenshot 2024-06-14 at 23 29 53

Screenshot 2024-06-14 at 00 19 07

### CODE REVIEW

File: misk-admin/src/main/kotlin/misk/web/metadata/config/ConfigMetadataAction.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ import misk.web.mediatype.MediaTypes
 class ConfigMetadataAction @Inject constructor(
   private val metadataProvider: Provider<ConfigMetadata>
 ) : WebAction {
-  @Get("/api/config/metadata")
+  @Get("/api/v1/config/metadata")
```
Comment (adrw) [inline misk-admin/src/main/kotlin/misk/web/metadata/config/ConfigMetadataAction.kt:17]: To maintain compatibility, both the old Misk Web UI still relying on this and the web actions have been migrated to a v1 prefix. This prevents and conflict with the new filterable `AllMetadataAction`

File: misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt [hunk 2]
```diff
@@ -11,15 +12,24 @@ import misk.web.metadata.Metadata
...
   @AllMetadataAccess
-  fun getAll(): Response {
-    return Response(all = metadata)
+  fun getMetadata(@PathParam id: String): Response {
+    // Return all metadata if the id is "all"
+    if (id == "all") return Response(metadata = allMetadata)
+
+    // Return metadata for the requested id
+    val metadata = allMetadata.filter { it.key == id }
+    return Response(metadata = metadata)
```
Comment (adrw) [inline misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt:27]: @brunofrts I updated the variable name from `all` to `metadata`

Comment (adrw) [reply misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt:27]: Open to suggestions if another variable name sounds better.

File: misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt [hunk 3]
```diff
@@ -11,15 +12,24 @@ import misk.web.metadata.Metadata
...
   }
 
-  data class Response(val all: Map<String, Metadata>)
+  data class Response(val metadata: Map<String, Metadata>)
+
+  companion object {
+    const val PATH = "/api/{id}/metadata"
```
Comment (adrw) [inline misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt:33]: Now takes a path param, `all` to get all metadata, or the metadata id to get a specific one (ie. `config`).

**Review by danieloh0714 — APPROVED** [review]

File: misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt (hunk 2 above)
Comment (adrw) [reply misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt]: I reverted this change.

**Pull request merged**